CIME/build_scripts/buildlib.kokkos

```diff
@@ -48,62 +50,102 @@ OR
 ###############################################################################
 def buildlib(bldroot, installpath, case):
     ###############################################################################
+    # If we are using Albany/Trilinos, we must use the Kokkos that comes with it
+    if case.get_value("USE_ALBANY") or case.get_value("USE_TRILINOS"):
+        print("case is using Trilinos and will use its Kokkos")
+        return
+
+    installed_kokkos_dir = os.environ.get("Kokkos_ROOT")
+    if installed_kokkos_dir is not None:
+        # We are trying to use a pre-installed kokkos. Look for the relevant folders/libs,
+        # and if all looks good, return. Otherwise, crap out
+        kokkos_root = os.path.abspath(installed_kokkos_dir)
+        include_dir = os.path.join(kokkos_root, "include")
+
+        expect(
+            os.path.isdir(kokkos_root),
+            f"Non-existent kokkos install dir '{kokkos_root}'",
+        )
+        expect(
+            os.path.isdir(include_dir),
+            f"Missing include subfolder in kokkos install dir '{kokkos_root}'",
+        )
+        expect(
+            os.path.isfile(os.path.join(include_dir, "Kokkos_Core.hpp")),
+            f"Missing kokkos headers in '{include_dir}'",
+        )
+
+        # TODO: should I check for libs too? The problem is that the lib subfolder is
+        # often arch dependent (e.g., $prefix/lib or $prefix/lib64)... The best thing
+        # would be to run a small cmake script, that calls find_package. But E3SM's
+        # cmake build system will do that soon enough, so any error will be caught there.
+        print(f"Using pre-installed Kokkos_ROOT: {kokkos_root}")
+        return
+    else:
+        print("no value found in env for Kokkos_ROOT. building from scratch")
+
     srcroot = case.get_value("SRCROOT")
-    kokkos_dir = os.path.join(srcroot, "externals", "kokkos")
-    expect(os.path.isdir(kokkos_dir), "Missing kokkos submodule")
+    cime_model = case.get_value("MODEL")
+    if cime_model == "e3sm":
+        ekat_dir = os.path.join(srcroot, "externals", "ekat")
+    else:
+        ekat_dir = os.path.join(srcroot, "libraries", "ekat")
```

Comment:
Can we remove the model check and just check which folder exists? @jedwards4b Would we encounter a case where both a present?

Reply:
Actually I think only CAM will use this script to build kokkos (E3SM should use a different one at https://github.com/sjsprecious/E3SM/blob/master/share/build/buildlib.kokkos). So I may just need to set the path for CAM here? @jedwards4b @jasonb5

Reply:
Yes, E3SM will never use this. It's just a minor design flaw to have CESM-specific stuff in CIME file.

Reply:
Thanks @jgfouca . I have moved this script to a CAM-specific place now.